internal/cluster/validations/validations.go

```diff
@@ -404,13 +404,63 @@ func ValidateIPAddresses(ipV6Supported bool, obj interface{}) error {
 	if err != nil {
 		return err
 	}
-	networkTypes := []string{"ClusterNetworks", "ServiceNetworks", "MachineNetworks"}
-	for _, net := range networkTypes {
-		networks := common.GetNetworkCidrAttr(obj, net)
-		err = ValidateDualStackIPNetworksOrder(networks...)
+	err = ValidateDualStackNetworks(obj)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ValidateDualStackNetworks(clusterParams interface{}) error {
```

Comment:
Does this need to be public/exported?

Reply:
Nope, it does not need. It's only like this because other validators in this file are also public even if they are used only in this package. Do you prefer to make this one private or to have them all the same ?

Reply:
Generally code is easier to maintain when there is less exporting, so I would go that direction. It's very easy to come back later and export it if/when the need arises.

Reply:
I agree with @mhrivnak and @djzager here. Let's make an attempt to default to private as much as possible. 

I would even be ok with a follow-up PR (that won't be backported) that makes the rest of these functions private (for the sake of consistency). Although, I would also understand it if people would rather not do such a refactor :)

Reply:
Fixed